scripts/data_overlap/get_group_overlap_stats.py

```diff
@@ -0,0 +1,91 @@
+import json
+import cattrs
+import os
+import argparse
+from typing import List, Dict
+
+from common.general import asdict_without_nones
+
+from data_overlap_spec import DataOverlapStats
+from light_scenario import GroupScenarioSpecs, GroupOverlapStats, AllGroupOverlapStats
+
+
+def save_group_overlap_stats_to_jsonl(all_group_overlap_stats: AllGroupOverlapStats, filename: str):
+    with open(filename, "a") as f:
+        f.write(json.dumps(asdict_without_nones(all_group_overlap_stats), ensure_ascii=False) + "\n")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--overlap_stats_path", type=str, required=True, help="The path to the overlap stats file")
+    parser.add_argument(
+        "--group_scenario_specs_path", type=str, required=True, help="The path to the group scenario specs file"
+    )
+    parser.add_argument("--output-data", type=str, required=True, help="The path to the output file")
+    parser.add_argument("--models", nargs="*", required=False, help="The name(s) of the model(s)")
+    args = parser.parse_args()
+
+    try:
+        os.remove(args.output_data)
+    except OSError:
+        pass
+
+    overlap_stats_jsons = open(args.overlap_stats_path, "r").readlines()
+
+    data_overlap_stats_list = []
+    for overlap_stats_json in overlap_stats_jsons:
+        overlap_stats_dict = json.loads(overlap_stats_json)
+        data_overlap_stats_list.append(cattrs.structure(overlap_stats_dict, DataOverlapStats))
+
+    scenario_spec_overlap_counts: Dict = dict()
+    for data_overlap_stats in data_overlap_stats_list:
+        data_overlap_stats_key = data_overlap_stats.data_overlap_stats_key
+        light_scenario_key = data_overlap_stats_key.light_scenario_key
+        scenario_spec = light_scenario_key.scenario_spec
+        num_instances = data_overlap_stats.num_instances
+        n = data_overlap_stats_key.overlap_protocol_spec.n
+        num_overlapping_inputs = len(data_overlap_stats.instance_ids_with_overlapping_input)
+        num_overlapping_references = len(data_overlap_stats.instance_ids_with_overlapping_reference)
+        if n == 13:
```

Comment:
Is there a way to avoid hardcoding?  At least add a comment for why this exists

Reply:
Added as optional arg default to 13